renku/api/repository.py

```diff
@@ -251,37 +254,44 @@ def subclients(self, parent_commit):
                 Submodule.iter_items(self.repo, parent_commit=parent_commit)
             ]
         except (RuntimeError, ValueError):
-            # There are no submodules assiciated with the given commit.
+            # There are no submodules associated with the given commit.
             submodules = []
 
-        return self._subclients.setdefault(
-            parent_commit, {
-                submodule: self.__class__(
-                    path=(self.path / submodule.path).resolve(),
+        subclients = {}
+        for submodule in submodules:
+            subpath = (self.path / submodule.path).resolve()
+            is_renku = subpath / Path(self.renku_home)
+
+            if subpath.exists() and is_renku.exists():
+                subclients[submodule] = self.__class__(
+                    path=subpath,
                     parent=(self, submodule),
                 )
-                for submodule in submodules
-            }
-        )
+
+        return subclients
 
     def resolve_in_submodules(self, commit, path):
         """Resolve filename in submodules."""
         original_path = self.path / path
-        if original_path.is_symlink() or str(path
-                                             ).startswith('.renku/vendors'):
+        in_vendor = str(path).startswith('.renku/vendors')
+
+        if original_path.is_symlink() or in_vendor:
```

Comment:
Why not resolve the path before and do the `in_vendor` check on the resolved path?

Reply:
This is just a rewrite for easier reading the code. Code is the same as before, just nicer to read.

Reply:
yep I know, I was just wondering about the order of operations